### PR TITLE
Endpoint for last clip

### PR DESCRIPTION
**Pull request by vanseforge**
I added a api endpoint for the last clip of an event (`/<camera_name>/<label>/last_clip.mp4`).
For example it can be used if the endpoint of the last thumbnail is used to open the clip.

### CODE REVIEW

File: frigate/http.py [hunk 1]
```diff
@@ -834,6 +834,22 @@ def label_thumbnail(camera_name, label):
...
 
+@bp.route("/<camera_name>/<label>/last_clip.mp4")
+def label_last_clip(camera_name, label):
+    label = unquote(label)
+    event_query = Event.select(fn.MAX(Event.id)).where(Event.camera == camera_name, Event.has_clip == True)
+    if label != "any":
+        event_query = event_query.where(Event.label == label)
+
+    try:
+        event = event_query.scalar()
```
Comment (NickM-27) [inline frigate/http.py:845]:
```suggestion
        event = event_query.get()
```

Comment (NickM-27) [reply frigate/http.py:845]: event ids are a string so scalar won't return correctly

File: frigate/http.py [hunk 2]
```diff
@@ -834,6 +834,22 @@ def label_thumbnail(camera_name, label):
         response.headers["Cache-Control"] = "no-store"
         return response
 
+@bp.route("/<camera_name>/<label>/last_clip.mp4")
+def label_last_clip(camera_name, label):
```
Comment (NickM-27) [inline frigate/http.py:838]:
```suggestion
def label_clip(camera_name, label):
```

File: frigate/http.py [hunk 3]
```diff
@@ -834,6 +834,22 @@ def label_thumbnail(camera_name, label):
         response.headers["Cache-Control"] = "no-store"
         return response
 
+@bp.route("/<camera_name>/<label>/last_clip.mp4")
```
Comment (NickM-27) [inline frigate/http.py:837]:
```suggestion
@bp.route("/<camera_name>/<label>/clip.mp4")
```

File: docs/docs/integrations/api.md [hunk 4]
```diff
@@ -237,6 +237,10 @@ Returns a thumbnail for the event id optimized for notifications. Works while th
 
 Returns the thumbnail from the latest event for the given camera and label combo. Using `any` as the label will return the latest thumbnail regardless of type.
 
+### `GET /api/<camera_name>/<label>/last_clip.mp4`
```
Comment (NickM-27) [inline docs/docs/integrations/api.md:240]:
```suggestion
### `GET /api/<camera_name>/<label>/clip.mp4`
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/http.py (hunk 1 above)
Comment (vanseforge) [reply frigate/http.py:845]: I used it like here:
https://github.com/blakeblackshear/frigate/blob/f6a4c2a7b3482acd9c787edf6dbb51898a1c021a/frigate/http.py#L825

Should this line than also be changed?

Comment (NickM-27) [reply frigate/http.py:845]: hmm, guess it's fine, I find it odd that that one works

**Review by vanseforge — COMMENTED** [review]
changed

**Review by NickM-27 — APPROVED** [review]

**Comment by vanseforge** [comment]
@NickM-27 I don't see what I have to do to get this merged. Can you please help me?
grafik

**Comment by NickM-27** [comment]
right now there is nothing left for you to do

**Comment by NickM-27** [comment]
python checks have failed, this needs to be fixed.

**Comment by vanseforge** [comment]
> python checks have failed, this needs to be fixed.

I tried to, but I don't understand what is wrong. Can you please explain it to me?
I don't see an error.
grafik

**Comment by NickM-27** [comment]
you need to run ruff to format the file

**Review by NickM-27 — APPROVED** [review]

**Pull request merged**